homeassistant/config.py

```diff
@@ -391,7 +355,6 @@ def _validate_currency(data: Any) -> Any:
                 _no_duplicate_auth_mfa_module,
             ),
             vol.Optional(CONF_MEDIA_DIRS): cv.schema_with_slug_keys(vol.IsDir()),
-            vol.Optional(CONF_LEGACY_TEMPLATES): cv.boolean,
```

Comment:
This is the core configuration, so we need to be careful; as this is the stuff that can trigger recovery mode.

What if someone still has is set? Will that render the configuration invalid? If so, should we add it back using `vol.Remove`?